components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -547,70 +589,104 @@ func (g *RTKMovementSensor) ReceiveAndWriteSerial() {
 
 // NtripStatus returns true if connection to NTRIP stream is OK, false if not.
 func (g *RTKMovementSensor) NtripStatus() (bool, error) {
+	g.mu.Lock()
+	defer g.mu.Unlock()
 	return g.ntripStatus, g.lastError
 }
 
 // Position returns the current geographic location of the MOVEMENTSENSOR.
 func (g *RTKMovementSensor) Position(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+	g.mu.Lock()
 	if g.lastError != nil {
+		defer g.mu.Unlock()
 		return &geo.Point{}, 0, g.lastError
 	}
+	g.mu.Unlock()
+
 	return g.nmeamovementsensor.Position(ctx, extra)
 }
 
 // LinearVelocity passthrough.
 func (g *RTKMovementSensor) LinearVelocity(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	g.mu.Lock()
```

Comment:
I believe if we have `mu` defined as a `RWMutex` instead, we can just use a read lock here (and in other places where we're just checking the `lastError` property). We might otherwise have some latency issues

Reply:
Gautham and I discussed: if the lock was held for a long time by multiple reading coroutines, we both agree that changing it from a `Mutex` to a `RWMutex` would be a great idea, and we also both agree that given how quickly the lock is released and how few coroutines will be using it, this isn't obviously an improvement. Neither of us feel strongly either way, so I'm going to leave it as-is until it becomes important.